components/brave_rewards/resources/ui/components/checkout/addFundsPanel/index.tsx

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import { Button } from 'brave-ui/components'
+import { BatColorIcon } from 'brave-ui/components/icons'
+
+import { FormSection } from '../formSection'
+import { CreditCardForm, CreditCardFormHandle, CreditCardDetails } from '../creditCardForm'
+import { TermsOfSale } from '../termsOfSale'
+import { GoBackLink } from '../goBackLink'
+
+import {
+  Subtitle,
+  CurrentBalance,
+  ExchangeRateDisplay,
+  AmountOptionContainer,
+  AmountOptionExchange,
+  ChargeSummary,
+  PurchaseButtonRow
+} from './style'
+
+interface AddFundsAmountOption {
+  amount: number
+  amountConverted: string
+  transactionFeeRate: string
+  transactionFee: string
+  totalCharge: string
+}
+
+interface AddFundsPanelProps {
+  amountNeeded: number
+  walletBalance: number
+  walletBalanceConverted: string
+  unitValueConverted: string
+  amountOptions: AddFundsAmountOption[]
+  onCancel: () => void
+  onPayWithCreditCard: (cardDetails: CreditCardDetails) => void
+}
+
+export function AddFundsPanel (props: AddFundsPanelProps) {
+  if (props.amountOptions.length === 0) {
+    throw new Error('props.amountOptions cannot be an empty array')
```

Comment:
I think we should not throw as it will display blank component. I think we should handle this more softly

Reply:
Done - if given an empty array, the component will simply not display any option buttons.